services/web/client/source/class/qxapp/desktop/PrjEditor.js

```diff
@@ -59,23 +61,24 @@ qx.Class.define("qxapp.desktop.PrjEditor", {
 
     this.showSettings(false);
 
-    this.__settingsView.addListener("SettingsEditionDone", function() {
+    this.__settingsView.addListener("SettingsEdited", function() {
       this.showSettings(false);
     }, this);
 
     this.__settingsView.addListener("ShowViewer", function(e) {
-      const metadata = e.getData().metadata;
-      const nodeId = e.getData().nodeId;
-      let url = "http://" + window.location.hostname + ":" + metadata.viewer.port;
-      if (metadata.viewer.entryPoint) {
-        url = url + "/" + metadata.viewer.entryPoint;
-      }
-      let viewerWin = this.__createBrowserWindow(url, metadata.name);
+      let data = e.getData();
+      let viewerWin = this.__createBrowserWindow(data.url, data.name);
+
+      //  const metadata = e.getData().metadata;
+      //  const nodeId = e.getData().nodeId;
+      //  let url = "http://" + window.location.hostname + ":" + metadata.viewer.port;
+      //  let viewerWin = this.__createBrowserWindow(url, metadata.name);
+
       this.__workbench.addWindowToDesktop(viewerWin);
 
       // Workaround for updating inputs
-      if (metadata.name === "3d-viewer") {
-        let urlUpdate = "http://" + window.location.hostname + ":" + metadata.viewer.port + "/retrieve";
+      if (data.name === "3d-viewer") {
+        let urlUpdate = "http://" + window.location.hostname + ":" + data.viewer.port + "/retrieve";
```

Comment:
we need to do something better for this. It should be somehow described by the service that this needs to be done prior to opening the service. Maybe create an issue for this.

Reply:
Agree

Reply:
yes ... this ties in with the whole reverse proxy topic